### PR TITLE
Fix compile warning for decals about using derivatives inside a loop

### PR DESCRIPTION
**Pull request by Paulm-Unity**

### CODE REVIEW

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Decal/DecalUtilities.hlsl [hunk 1]
```diff
@@ -75,21 +76,22 @@ void AddDecalContribution(PositionInputs posInput, inout SurfaceData surfaceData
 			int diffuseIndex = decalData.normalToWorld[1][3];
 			int normalIndex = decalData.normalToWorld[2][3];
 			int maskIndex = decalData.normalToWorld[3][3];
+			float lod = ComputeTextureLOD(positionDS.xz);
```
[the remark this answers is not in the document]
Comment (Paulm-Unity) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Decal/DecalUtilities.hlsl]: The only thing I checked was that the warning went away. I will try a test with different colors for each mip, maybe also look in Razor at the shader code it generates.

[the remark this answers is not in the document]
Comment (Paulm-Unity) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Decal/DecalUtilities.hlsl]: You are right it is inside the loop, but outside the dynamic branches. So maybe that is the reason it is not complaining. But I will do some further tests.

[the remark this answers is not in the document]
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Decal/DecalUtilities.hlsl]: Ah ok, the loop is a dynamic branch but it this context maybe the compiler is able to figure out that decal count is the same for a 2x2 quad (as it is the same for a tile of 16x16), so there is no problem. Ok, will just be good to make a test on the mips but now I get why it should be fine :)

**Comment by Paulm-Unity** [comment]
Verified that lod calculation works with a test texture, looking at the shader code in Razor the lod calculation is outside the dynamic branches where the texture fetch happens, however compiler was placing it inside the branch that was rejecting the pixel, so had to use alpha blend of 0 instead of branching to solve that issue.

**Pull request merged**